<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>gui::MainWindow</class>
 <widget class="QMainWindow" name="gui::MainWindow">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>857</width>
    <height>733</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="focusPolicy">
   <enum>Qt::StrongFocus</enum>
  </property>
  <property name="acceptDrops">
   <bool>true</bool>
  </property>
  <property name="windowTitle">
   <string>Vistle[*]</string>
  </property>
  <property name="documentMode">
   <bool>true</bool>
  </property>
  <property name="dockNestingEnabled">
   <bool>false</bool>
  </property>
  <property name="dockOptions">
   <set>QMainWindow::AllowTabbedDocks|QMainWindow::AnimatedDocks|QMainWindow::ForceTabbedDocks</set>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout_3">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="gui::DataFlowView" name="drawArea">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>10</horstretch>
        <verstretch>10</verstretch>
       </sizepolicy>
      </property>
      <property name="cursor" stdset="0">
       <cursorShape>CrossCursor</cursorShape>
      </property>
      <property name="focusPolicy">
       <enum>Qt::NoFocus</enum>
      </property>
      <property name="acceptDrops">
       <bool>true</bool>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <property name="lineWidth">
       <number>0</number>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>857</width>
     <height>24</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuHello_World">
    <property name="title">
     <string>Fi&amp;le</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpenOnGui"/>
    <addaction name="actionOpenOnHub"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveOnGui"/>
    <addaction name="actionSaveOnHub"/>
    <addaction name="separator"/>
    <addaction name="actionShow_Session_URL"/>
    <addaction name="actionCopy_Session_URL"/>
    <addaction name="separator"/>
    <addaction name="actionConnect"/>
    <addaction name="actionQuit"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionSelectClear"/>
    <addaction name="actionSelectInvert"/>
    <addaction name="actionSelect_All"/>
    <addaction name="separator"/>
    <addaction name="actionDelete"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
    <addaction name="actionAbout_License"/>
    <addaction name="actionAbout_Icons"/>
    <addaction name="actionAbout_Qt"/>
   </widget>
   <widget class="QMenu" name="menuModules">
    <property name="title">
     <string>Modules</string>
    </property>
    <addaction name="actionViewOrig"/>
    <addaction name="actionViewAll"/>
    <addaction name="actionSnapToGrid"/>
    <addaction name="separator"/>
    <addaction name="actionSelectSources"/>
    <addaction name="actionSelectSinks"/>
    <addaction name="separator"/>
    <addaction name="actionExecute"/>
   </widget>
   <addaction name="menuHello_World"/>
   <addaction name="menuEdit"/>
   <addaction name="menuModules"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QDockWidget" name="consoleDock">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>106</width>
     <height>54</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>524287</width>
     <height>524287</height>
    </size>
   </property>
   <property name="focusPolicy">
    <enum>Qt::NoFocus</enum>
   </property>
   <property name="allowedAreas">
    <set>Qt::BottomDockWidgetArea|Qt::TopDockWidgetArea</set>
   </property>
   <property name="windowTitle">
    <string>Vistle Console</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="gui::VistleConsole" name="consoleWidget">
    <layout class="QHBoxLayout" name="horizontalLayout_2"/>
   </widget>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>Toolbar</string>
   </property>
   <property name="movable">
    <bool>false</bool>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonTextBesideIcon</enum>
   </property>
   <property name="floatable">
    <bool>false</bool>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionOpen"/>
   <addaction name="actionSave"/>
   <addaction name="separator"/>
   <addaction name="actionExecute"/>
   <addaction name="separator"/>
   <addaction name="actionSelectInvert"/>
   <addaction name="actionSelect_All"/>
   <addaction name="actionDelete"/>
   <addaction name="separator"/>
   <addaction name="actionViewOrig"/>
   <addaction name="actionViewAll"/>
   <addaction name="actionArrange"/>
  </widget>
  <widget class="QDockWidget" name="modulesDock">
   <property name="sizePolicy">
    <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="focusPolicy">
    <enum>Qt::StrongFocus</enum>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="allowedAreas">
    <set>Qt::LeftDockWidgetArea|Qt::RightDockWidgetArea</set>
   </property>
   <property name="windowTitle">
    <string>Module Browser</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="gui::ModuleBrowser" name="moduleBrowser">
    <property name="focusPolicy">
     <enum>Qt::StrongFocus</enum>
    </property>
   </widget>
  </widget>
  <widget class="QDockWidget" name="moduleViewDock">
   <property name="focusPolicy">
    <enum>Qt::StrongFocus</enum>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="allowedAreas">
    <set>Qt::LeftDockWidgetArea|Qt::RightDockWidgetArea</set>
   </property>
   <property name="windowTitle">
    <string>Module</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="gui::ModuleView" name="moduleView">
    <property name="focusPolicy">
     <enum>Qt::StrongFocus</enum>
    </property>
   </widget>
  </widget>
  <action name="actionNew">
   <property name="text">
    <string>&amp;New</string>
   </property>
   <property name="toolTip">
    <string>Remove all modules from workflow</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="icon">
    <iconset resource="gui.qrc">
     <normaloff>:/icons/open.svg</normaloff>:/icons/open.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Open...</string>
   </property>
   <property name="toolTip">
    <string>Open existing workflow</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="gui.qrc">
     <normaloff>:/icons/save.svg</normaloff>:/icons/save.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Save</string>
   </property>
   <property name="toolTip">
    <string>Save workflow</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>&amp;Quit</string>
   </property>
   <property name="toolTip">
    <string>Quit</string>
   </property>
  </action>
  <action name="actionCut">
   <property name="text">
    <string>&amp;Cut</string>
   </property>
  </action>
  <action name="actionCopy">
   <property name="text">
    <string>C&amp;opy</string>
   </property>
  </action>
  <action name="actionPaste">
   <property name="text">
    <string>&amp;Paste</string>
   </property>
  </action>
  <action name="actionSelect_All">
   <property name="icon">
    <iconset resource="gui.qrc">
     <normaloff>:/icons/select.svg</normaloff>:/icons/select.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Select All</string>
   </property>
   <property name="toolTip">
    <string>Select all modules</string>
   </property>
  </action>
  <action name="actionDelete">
   <property name="icon">
    <iconset resource="gui.qrc">
     <normaloff>:/icons/delete.svg</normaloff>:/icons/delete.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Delete</string>
   </property>
   <property name="toolTip">
    <string>Delete selected modules</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="text">
    <string>&amp;Undo</string>
   </property>
  </action>
  <action name="actionSaveOnGui">
   <property name="text">
    <string>Sa&amp;ve Locally...</string>
   </property>
   <property name="toolTip">
    <string>Save workflow as a new file on GUI</string>
   </property>
  </action>
  <action name="actionSettings">
   <property name="text">
    <string>S&amp;ettings</string>
   </property>
  </action>
  <action name="actionArrange">
   <property name="visible">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;Arrange</string>
   </property>
   <property name="toolTip">
    <string>Automatically arrange modules</string>
   </property>
  </action>
  <action name="actionConnect">
   <property name="text">
    <string>&amp;Connect</string>
   </property>
   <property name="toolTip">
    <string>Connect to Vistle</string>
   </property>
  </action>
  <action name="actionExecute">
   <property name="icon">
    <iconset resource="gui.qrc">
     <normaloff>:/icons/execute.svg</normaloff>:/icons/execute.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Execute Workflow</string>
   </property>
   <property name="toolTip">
    <string>Execute all source modules</string>
   </property>
  </action>
  <action name="actionNative_Menubar">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Native Menubar</string>
   </property>
   <property name="toolTip">
    <string>Use native menubar</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
   <property name="toolTip">
    <string>About Vistle</string>
   </property>
  </action>
  <action name="actionAbout_License">
   <property name="text">
    <string>License</string>
   </property>
  </action>
  <action name="actionAbout_Icons">
   <property name="text">
    <string>About Icons</string>
   </property>
  </action>
  <action name="actionAbout_Qt">
   <property name="text">
    <string>About Qt</string>
   </property>
  </action>
  <action name="actionShow_Session_URL">
   <property name="text">
    <string>Show Session URL</string>
   </property>
  </action>
  <action name="actionCopy_Session_URL">
   <property name="text">
    <string>Copy Session URL</string>
   </property>
  </action>
  <action name="actionViewAll">
   <property name="icon">
    <iconset resource="gui.qrc">
     <normaloff>:/icons/viewall.svg</normaloff>:/icons/viewall.svg</iconset>
   </property>
   <property name="text">
    <string>View All</string>
   </property>
   <property name="toolTip">
    <string>Zoom to whole workflow graph</string>
   </property>
  </action>
  <action name="actionViewOrig">
   <property name="icon">
    <iconset resource="gui.qrc">
     <normaloff>:/icons/vieworig.svg</normaloff>:/icons/vieworig.svg</iconset>
   </property>
   <property name="text">
    <string>Original Size</string>
   </property>
   <property name="toolTip">
    <string>Zoom workflow graph to original size</string>
   </property>
  </action>
  <action name="actionSelectInvert">
   <property name="icon">
    <iconset resource="gui.qrc">
     <normaloff>:/icons/invert.svg</normaloff>:/icons/invert.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Invert Selection</string>
   </property>
   <property name="toolTip">
    <string>Invert selection</string>
   </property>
  </action>
  <action name="actionSelectSources">
   <property name="text">
    <string>&amp;Select Sources</string>
   </property>
   <property name="toolTip">
    <string>Select source modules</string>
   </property>
  </action>
  <action name="actionSelectSinks">
   <property name="text">
    <string>&amp;Select Sinks</string>
   </property>
   <property name="toolTip">
    <string>Select sink modules</string>
   </property>
  </action>
  <action name="actionSelectClear">
   <property name="text">
    <string>&amp;Clear Selection</string>
   </property>
   <property name="toolTip">
    <string>Deselect selected modules</string>
   </property>
  </action>
  <action name="actionSnapToGrid">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Snap To Grid</string>
   </property>
   <property name="toolTip">
    <string>snap modules to positions aligning ports</string>
   </property>
  </action>
  <action name="actionSaveOnHub">
   <property name="text">
    <string>Sa&amp;ve On Hub...</string>
   </property>
   <property name="toolTip">
    <string>Save workflow as a new file on hub</string>
   </property>
  </action>
  <action name="actionOpenOnHub">
   <property name="icon">
    <iconset resource="gui.qrc">
     <normaloff>:/icons/open.svg</normaloff>:/icons/open.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Open On Hub...</string>
   </property>
   <property name="toolTip">
    <string>Open existing workflow on hub</string>
   </property>
  </action>
  <action name="actionOpenOnGui">
   <property name="icon">
    <iconset resource="gui.qrc">
     <normaloff>:/icons/open.svg</normaloff>:/icons/open.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Open Locally...</string>
   </property>
   <property name="toolTip">
    <string>Open existing workflow on GUI</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>gui::DataFlowView</class>
   <extends>QGraphicsView</extends>
   <header>dataflowview.h</header>
  </customwidget>
  <customwidget>
   <class>gui::ModuleBrowser</class>
   <extends>QWidget</extends>
   <header>modulebrowser.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>gui::VistleConsole</class>
   <extends>QWidget</extends>
   <header>vistleconsole.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>gui::ModuleView</class>
   <extends>QWidget</extends>
   <header>moduleview.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="gui.qrc"/>
 </resources>
 <connections/>
</ui>
